bin/amphora/src/amphora/node/stringlock.rs

```diff
@@ -0,0 +1,129 @@
+use std::mem;
+use std::sync::Arc;
+use std::time::Instant;
+use std::{collections::HashMap, time::Duration};
+
+use tokio::sync::{Mutex, MutexGuard, RwLock};
+
+static TOKIO_MUTEX_SIZE: usize = mem::size_of::<Mutex<()>>();
+
+struct StringLockEntry {
+    lock: Arc<RwLock<()>>,
+    last_accessed: Instant,
+}
+
+impl Default for StringLockEntry {
+    fn default() -> Self {
+        Self {
+            lock: Default::default(),
+            last_accessed: Instant::now(),
+        }
+    }
+}
+
+pub struct StringLock {
+    data: Mutex<HashMap<String, StringLockEntry>>,
+    lifetime: Duration,
+    cleanup_trigger: Option<usize>,
+}
+
+impl StringLock {
+    pub fn new(lifetime: Duration) -> Self {
+        Self {
+            data: Default::default(),
+            lifetime,
+            cleanup_trigger: None,
+        }
+    }
+
+    pub fn with_cleanup_trigger(mut self, max_memory: usize) -> Self {
```

Comment:
Why not use a count instead of a memory size given the size of the mutex is constant?

Reply:
Parce que 1- le size est constant _pour une plateforme donnée_. Ça peut varier d'une plateforme à l'autre. Donc en faisant ça on s'assure que l'overhead pour les mutex est constant pour toutes les plateformes.

Reply:
Ahhh I see!